packages/shared/src/hooks/post/useTrackedCopyPostLink.ts

```diff
@@ -0,0 +1,23 @@
+import { useContext } from 'react';
+import { Post } from '../../graphql/posts';
+import { useCopyPostLink } from '../useCopyPostLink';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+import { ShareProvider } from '../../lib/share';
+import { postAnalyticsEvent } from '../../lib/feed';
+import { Origin } from '../../lib/analytics';
+
+type UseTrackedCopyPostLink = (provider: ShareProvider) => void;
+
+export const useTrackedCopyPostLink = (post: Post): UseTrackedCopyPostLink => {
```

Comment:
I could have sworn we already had this hook 😓 
(Can't find it, but guess we at sometime need to clean up our hooks a bit)

Reply:
Yep. We also need to start organizing files into folders 😅

Reply:
We are using this hook only here now, do we intend to replace other usages with this hook?

Reply:
I think so, but we should be careful when we do refactoring of events as it is a very delicate area.